gcs/src/gcs.cpp

```diff
@@ -199,6 +199,20 @@ struct gcs_conn
     int outer_close_count; // how many times gcs_close has been called.
 };
 
+gcs_node_state_t gcs_get_state_for_idx(gcs_conn_t* conn, ssize_t idx) {
+  const gcs_group_t* group = gcs_core_get_group(conn->core);
+  if (idx >= group->num) {
+    // XXX: Dirty fix.
```

Comment:
This looks like a real issue that could also happen for 8.0, we just didn't notice it yet - and we do need this condition here as a safeguard.

On the other hand, it is not the full fix for this, I'll create a separate jira issue about that for us, and this condition is good enough for now, you can remove the comment from it.

Reply:
Ok, but in fairness, for some reason this segfault is not reproduced in 8.0.
Maybe it would be better to use optional (like boost optional or something)? Or introduce new GCS_NODE_STATE_* status. GCS_NODE_STATE_MAX seems not to be a good choice here.

Reply:
The main issue here is that we keep using the `idx` we obtained at the start of the SST request, and there's no guarantee that the node array doesn't change during it. That's also true for 8.0.

For now this fix, an optional could work, yes, and would be less of a hack.

Reply:
Done